doc/hardware/porting/custom.rst

```diff
@@ -0,0 +1,16 @@
+.. _platform_cusomizations:
+
+Hardware with Custom Interfaces
```

Comment:
Seems like a good first step towards closing the documentation gap between "board porting" and "soc" or "arch" porting.